.github/workflows/ci-docs-tests.yml

```diff
@@ -0,0 +1,154 @@
+
+      - name: "conda install"
+        uses: conda-incubator/setup-miniconda@v2
+        with:
+          miniforge-version: latest
+          channels: conda-forge,defaults
+          activate-environment: ${{ env.ENV_NAME }}
+          auto-update-conda: false
+          use-only-tar-bz2: true
+
+      - name: "conda environment cache"
+        uses: actions/cache@v3
+        id: conda-env-cache
+        env:
+          # Increment the build number to force a cache refresh.
+          CACHE_BUILD: 0
+        with:
+          path: ${{ env.CONDA }}/envs/${{ env.ENV_NAME }}
+          key: ${{ runner.os }}-conda-env-${{ env.ENV_NAME }}-p${{ env.CACHE_PERIOD }}-b${{ env.CACHE_BUILD }}
+
+      - name: "conda environment update"
+        if: steps.conda-env-cache.outputs.cache-hit != 'true'
+        run: |
+          conda install --quiet --name ${{ env.ENV_NAME }} cartopy nox pip
+
+      - name: "conda info"
+        run: |
+          conda info
+          conda list
+
+      - name: "cartopy cache"
+        uses: actions/cache@v3
+        id: cartopy-cache
+        env:
+          # Increment the build number to force a cache refresh.
+          CACHE_BUILD: 0
+        with:
+          path: ~/.local/share/cartopy
+          key: ${{ runner.os }}-cartopy-${{ env.ENV_NAME }}-p${{ env.CACHE_PERIOD }}-b${{ env.CACHE_BUILD }}
+
+      - name: "cartopy cache populate"
+        if: steps.cartopy-cache.outputs.cache-hit != 'true'
+        env:
+          CARTOPY_SHARE_DIR: ~/.local/share/cartopy
+          CARTOPY_FEATURE: https://raw.githubusercontent.com/SciTools/cartopy/v0.20.0/tools/cartopy_feature_download.py
+        run: |
+          wget --quiet ${CARTOPY_FEATURE}
+          mkdir -p ${CARTOPY_SHARE_DIR}
+          # Requires a pre-installed version of cartopy within the environment.
+          python cartopy_feature_download.py physical --output ${CARTOPY_SHARE_DIR} --no-warn
+
+      - name: "nox cache"
+        uses: actions/cache@v3
+        env:
+          # Increment the build number to force a cache refresh.
+          CACHE_BUILD: 0
+        with:
+          path: ${{ github.workspace }}/.nox
+          key: ${{ runner.os }}-nox-${{ env.ENV_NAME }}-py${{ matrix.python-version }}-b${{ env.CACHE_BUILD }}-${{ hashFiles(env.LOCK_FILE) }}
+
+      - name: "iris configure"
+        env:
+          SITE_CFG: lib/iris/etc/site.cfg
+          MPL_RC: ${HOME}/.config/matplotlib/matplotlibrc
+        run: |
+          mkdir -p $(dirname ${SITE_CFG})
+          echo ${SITE_CFG}
+          echo "[Resources]" >> ${SITE_CFG}
+          echo "test_data_dir = ${HOME}/iris-test-data/test_data" >> ${SITE_CFG}
+          echo "doc_dir = ${GITHUB_WORKSPACE}/docs" >> ${SITE_CFG}
+          cat ${SITE_CFG}
+          mkdir -p $(dirname ${MPL_RC})
+          echo ${MPL_RC}
+          echo "backend : agg" >> ${MPL_RC}
+          echo "image.cmap : viridis" >> ${MPL_RC} 
+          cat ${MPL_RC}
```

Comment:
I'd like to see this whole section disappear, and similar with the other GHAs... but not in the PR.

Save that for another day.

Reply:
Done